oak/common/app_config.cc

```diff
@@ -153,6 +156,45 @@ bool AddStorageToConfig(ApplicationConfiguration* config, const std::string& nam
   return false;
 }
 
+bool AddEscrowToConfig(ApplicationConfiguration* config, const std::string& name,
+                        const std::string& storage_address) {
+  for (auto& node : *config->mutable_nodes()) {
+    if (!node.has_storage_node() || node.node_name() != name) {
+      continue;
+    }
+    Node* escrow_node = config->add_nodes();
+    escrow_node->set_node_name(kEscrowNodeName);
+    escrow_node->mutable_escrow_node()->set_address(storage_address);
+
+    StorageProxyNode* storage_node = node.mutable_storage_node();
+    Port* out_port = storage_node->add_ports();
+    out_port->set_name(kEscrowOutPortName);
+    out_port->set_type(Port_Type_OUT);
+    Port* in_port = storage_node->add_ports();
+    in_port->set_name(kEscrowInPortName);
+    in_port->set_type(Port_Type_IN);
+
+    Channel* out_channel = config->add_channels();
+    Channel_Endpoint* out_src = out_channel->mutable_source_endpoint();
+    out_src->set_node_name(node.node_name());
+    out_src->set_port_name(out_port->name());
+    Channel_Endpoint* out_dest = out_channel->mutable_destination_endpoint();
+    out_dest->set_node_name(escrow_node->node_name());
+    out_dest->set_port_name(kEscrowNodeRequestPortName);
+
+    Channel* in_channel = config->add_channels();
+    Channel_Endpoint* in_src = in_channel->mutable_source_endpoint();
+    in_src->set_node_name(escrow_node->node_name());
+    in_src->set_port_name(kEscrowNodeResponsePortName);
+    Channel_Endpoint* in_dest = in_channel->mutable_destination_endpoint();
+    in_dest->set_node_name(node.node_name());
+    in_dest->set_port_name(in_port->name());
```

Comment:
This seems to add a pair of channels between the storage pseudo-Node and the new escrow pseudo-Node; do we expect there to be other, different, channels to the escrow pseudo-Node in the future?  (If the escrow node is only ever connected to the storage node, then would it be simpler to just have a `bool uses_escrow` field on the storage pseudo-Node config?)

Reply:
Yeah I think actually the current design is problematic, we definitely don't want developers to connect a random node to the escrow and get secrets out of it! Secrets should only ever move in a parallel domain that is disjoint from the domain where data moves, i.e. we should *not* use regular channels to get secrets in and out of the escrow (unless perhaps we had a policy to tag TCB-level secrets, but that's not the case right now).
Do let me know if I misunderstood the design though.